src/utils/userLocation.ts

```diff
@@ -1,52 +1,83 @@
-import firebase, { UserInfo } from "firebase/app";
+import firebase from "firebase/app";
 
 import { AnyRoom } from "types/rooms";
 
 import { updateUserProfile } from "pages/Account/helpers";
 import { useInterval } from "hooks/useInterval";
 
 import { getCurrentTimeInMilliseconds } from "./time";
-import { openRoomUrl, openUrl, venueInsideUrl } from "./url";
+import { openRoomUrl } from "./url";
 
 const LOCATION_INCREMENT_SECONDS = 10;
 const LOCATION_INCREMENT_MS = LOCATION_INCREMENT_SECONDS * 1000;
 
 type LocationData = Record<string, number>;
 
-export const updateLocationData = (
-  userId: string,
-  newLocationData: LocationData = {}
-) => {
+type UpdateLocationDataProps = {
+  userId: string;
+  newLocationData: LocationData;
+};
+
+export const updateLocationData = ({
+  userId,
+  newLocationData,
+}: UpdateLocationDataProps) => {
   updateUserProfile(userId, {
     lastSeenAt: getCurrentTimeInMilliseconds(),
     lastSeenIn: newLocationData,
   });
 };
 
-export interface TrackLocationProps {
+type SetNewLocationDataProps = {
   userId: string;
   locationName: string;
   lastSeenIn?: LocationData;
-}
+};
 
-export const trackLocationEntered = ({
+export const setNewLocationData = ({
   userId,
   locationName,
-}: TrackLocationProps) => {
-  updateLocationData(userId, {
-    [locationName]: getCurrentTimeInMilliseconds(),
+}: SetNewLocationDataProps) => {
+  updateLocationData({
+    userId,
+    newLocationData: {
+      [locationName]: getCurrentTimeInMilliseconds(),
+    },
   });
 };
 
-export const enterVenue = (venueId: string) => openUrl(venueInsideUrl(venueId));
+type UpdateCurrentLocationDataProps = {
+  userId: string;
+  profileLocationData: LocationData;
+};
+
+export const updateCurrentLocationData = ({
+  userId,
+  profileLocationData,
+}: UpdateCurrentLocationDataProps) => {
+  const [locationName] = Object.keys(profileLocationData);
+
+  updateLocationData({
+    userId,
+    newLocationData: { [locationName]: getCurrentTimeInMilliseconds() },
+  });
+};
+
+type ClearLocationData = {
+  userId: string;
+};
+
+export const clearLocationData = ({ userId }: ClearLocationData) => {
```

Comment:
Since we only have a single parameter, it's probably fine to just make it `userId: string` directly, and not wrap it in object destructuring. Which also means we can get rid of the `ClearLocationData` type above.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1129/commits/53f57c40e24983470b7552d4ebed6f1715ff74ba